receiver/prometheusreceiver/internal/metricfamily.go

```diff
@@ -77,6 +86,40 @@ func newMetricFamily(metricName string, mc MetadataCache) MetricFamily {
 	}
 }
 
+// Define manualy the metadata of prometheus scrapper internal metrics
+func defineInternalMetric(metricName string, metadata scrape.MetricMetadata, logger *zap.Logger) scrape.MetricMetadata {
+	if metadata.Metric != "" && metadata.Type != "" && metadata.Unit != "" && metadata.Help != "" {
+		logger.Debug("Internal metric seems already fully defined")
+		return metadata
+	}
+	metadata.Metric = metricName
+
+	switch metricName {
+	case scrapeUpMetricName:
+		metadata.Unit = "bool"
+		metadata.Type = textparse.MetricTypeGauge
```

Comment:
The "up" metric coming from prometheus has no metadata (I queried the `api/v1/targets/metadata` endpoint on my prom server), but we have to pick something here.

It seems like the best fit might be a [StateSet](https://github.com/OpenObservability/OpenMetrics/blob/main/specification/OpenMetrics.md#stateset).  WDYT @Aneurysm9?

Reply:
The topic of StateSets came up in the metrics data model SIG the other day and I think we still don't have a plan for supporting them in OTLP.  See https://github.com/open-telemetry/opentelemetry-specification/issues/1712.  I think for the current purpose a Gauge is fine.